llm/server.go

```diff
@@ -217,15 +218,92 @@ func NewLlamaServer(gpus discover.GpuInfoList, model string, ggml *GGML, adapter
 		params = append(params, "--threads", strconv.Itoa(defaultThreads))
 	}
 
-	flashAttnEnabled := envconfig.FlashAttention()
+	// isEmbeddingModel checks for common GGML attributes that help distinguish most embedding models from normal models.
+	isEmbeddingModel := false
+	if _, ok := ggml.KV()[fmt.Sprintf("%s.pooling_type", ggml.KV().Architecture())]; ok {
+		isEmbeddingModel = true
+	}
+
+	// Validates and applies KV cache parameters
+	setCacheTypeParam := func(paramName, cacheType string) {
```

Comment:
Would it be possible to make this a validation function vs one that mutates `params` (that could happen on around line 294 below)

Reply:
Moved to a newly created llm/validation.go :)